src/main/java/com/yurjinia/common/handlers/LogoutHandler.java

```diff
@@ -0,0 +1,40 @@
+package com.yurjinia.common.handlers;
+
+import com.yurjinia.common.exception.CommonException;
+import com.yurjinia.common.exception.ErrorCode;
+import com.yurjinia.common.security.jwt.constants.JwtConstants;
+import com.yurjinia.common.security.jwt.service.JwtBlacklistService;
+import com.yurjinia.common.security.jwt.service.JwtService;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+@Component
+@RequiredArgsConstructor
+public class LogoutHandler {
+
+    private final JwtService jwtService;
+    private final JwtBlacklistService jwtBlacklistService;
+
+    public void logout(HttpServletRequest request, HttpServletResponse response, Authentication authentication) {
+        String requestHeader = request.getHeader(JwtConstants.AUTHORIZATION_HEADER);
+
+        if (StringUtils.hasText(requestHeader) && requestHeader.startsWith(JwtConstants.TOKEN_PREFIX)) {
```

Comment:
I'm thinking here we want to throw an exception in case this if returns false.

So first of all do the opposite check -> `if doesn't have text in header OR doesn't start with TOKEN_PREFIX` - throw an exception, `else - get token, get expiration time, check if blacklisted, blacklist token and then logout`.

Reply:
And also - there was a code `SecurityContextHolder.getContext().setAuthentication(null);`, is it not needed anymore?